src/scenes/AcceleratorRouter/CohortForm.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import { Container, Grid, useTheme } from '@mui/material';
+import { Dropdown, Textfield } from '@terraware/web-components';
+import PageForm from 'src/components/common/PageForm';
+import strings from 'src/strings';
+import { CreateCohortRequestPayload, UpdateCohortRequestPayload } from 'src/types/Cohort';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+
+type CohortFormProps<T extends CreateCohortRequestPayload | UpdateCohortRequestPayload> = {
+  busy?: boolean;
+  cohort: T;
+  onCancel: () => void;
+  onSave: (cohort: T) => void;
+};
+
+export default function CohortForm<T extends CreateCohortRequestPayload | UpdateCohortRequestPayload>(
+  props: CohortFormProps<T>
+): JSX.Element {
+  const { busy, cohort, onCancel, onSave } = props;
+
+  const { isMobile } = useDeviceInfo();
+  const theme = useTheme();
+
+  const [localRecord, setLocalRecord] = useState<T>(cohort);
+  const [validateFields, setValidateFields] = useState<boolean>(false);
+
+  const currentPhaseDropdownOptions = [
```

Comment:
useMemo with a dependency on activeLocale (in case user changes the language). Example: https://github.com/terraware/terraware-web/blob/main/src/components/AddNewOrganizationModal.tsx#L65